tests/modules/programs/vim/install.nix

```diff
@@ -0,0 +1,36 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+{
+  config = {
+    programs.vim = {
+      enable = true;
+
+      settings = { background = "dark"; };
+    };
+
+    nmt.script = ''
+      function assertAbsoluteFileExists() {
+        if [[ ! -f "$1" ]]; then
+          fail "Expected $1 to exist but it was not found."
+        fi
+      }
+
+      function assertAbsoluteFileContains() {
+        if ! grep -qF "$2" "$1"; then
+          fail "Expected $1 to contain $2 but it did not."
+        fi
+      }
+
+
+      assertFileExists home-path/bin/vim
+      assertFileIsExecutable home-path/bin/vim
+
+      rc_file=$(find /nix/store/ -type f -iname '*-vimrc' -not -iname '*nixos*' | tail -n1)
```

Comment:
I guess that doing this might take a long time depending how much one has in the nix store. Maybe it's better to get the generated vimrc file from the `vim` executable? See the example output of `$ cat home-path/bin/vim`:

```
#!/nix/store/hrpvwkjz04s9i4nmli843hyw9z4pwhww-bash-4.4-p23/bin/bash
exec /nix/store/frrf0ynmv0ald054dmbf276nyghnx9fv-vim_configurable-8.2.0013/bin/vim -u /nix/store/qzyarlqfxm5bwgrbk7d8vgjmnkqkgyl2-vimrc  "$@"
```